### PR TITLE
add complete sleep and quality for a week

### LINKED ISSUE #17: Create Sleep methods
- For a user (identified by their userID), the average number of hours slept per day
- For a user, their average sleep quality per day over all time
- For a user, how many hours they slept for a specific day (identified by a date)
- For a user, their sleep quality for a specific day (identified by a date)
- For a user, how many hours slept each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- For a user, their sleep quality each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- For all users, the average sleep quality
### Sleep.js
- [ ] `const sleep = new Sleep(data)`
### Properties
- [ ]  `this.id = data.id`
- [ ] `this.date = data.date`
- [ ] `this.sleepQuility = data.sleepQuility`

### Methods 

- [ ] `calAvergeSleptHoursInDay(id)`
- [ ] `calAllSleptHoursForUser()`
- [ ] `getHoursSlept(date)`
- [ ] `getQuelitySlept(date)`
- [ ] `calWeeklySleptHours(startDat, endDate)`
- [ ] `calWeeklyQualitySleptHours(startDat, endDate)`
- [ ] `calAllUsersAverageSleep()`

### LINKED ISSUE #18: Write unit tests for Sleep methods

### PR DESCRIPTION
**Pull request by piperjarrett**
What does this PR do?
Adds the methods to show how many hours slept each day and quality over the course of a given week (7 days).
Where to start?
User.js and USer-test.js
How was it tested? 
Passed all tests in the terminal
close #17 
close #18

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
+    }, 0);
+    const average = (totalSleepData / usersData.length).toFixed(1);
+    return parseFloat(average);
+  }
 
+  getSleepPerDayForWeek(sleepData, startDate, detail) {
+    const userSleep = sleepData.filter((entry) => entry.userID === this.id);
+    const startingDate = userSleep.find((entry) => entry.date === startDate);
+    const startIndex = userSleep.indexOf(startingDate);
+    let newArr = userSleep.slice(startIndex);
```
Comment (forsethnico) [inline src/User.js]: I think you could try to use the starting index and then a second parameter for the slice function, like startingIndex+7 perhaps. For example, slice(1,4) extracts the second element through the fourth element (elements indexed 1, 2, and 3). I haven't tested this out but it was kind of where my thoughts went.

File: src/User.js [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
+    const weekSleep = newArr.reduce((sleepForWeek, entry) => {
+      if (sleepForWeek.length < 7) {
```
Comment (forsethnico) [inline src/User.js]: What if sleepForWeek.length is equal to 7? What happens then?

File: test/User-test.js [hunk 3]
```diff
@@ -79,7 +81,44 @@ let user;
...
-  })
+    expect(
+      fakeUser.getAvgSleepDataPerDay(sleepTestData, "sleepQuality")
+    ).to.equal(0);
+    expect(
+      fakeUser.getAvgSleepDataPerDay(sleepTestData, "hoursSlept")
+    ).to.equal(0);
+  });
+  it("should have a method to calculate the hours slept per day over a week for a user and quality of sleep ", () => {
+    let sleeptestData;
```
Comment (forsethnico) [inline test/User-test.js]: This variable is not needed or used anywhere I don't believe.

File: src/User.js (hunk 2 above)
Comment (piperjarrett) [reply src/User.js]: it goes to the else and just returns the complete 7 day array.

File: src/User.js (hunk 1 above)
Comment (piperjarrett) [reply src/User.js]: Okay, I will try and see what happens.

File: test/User-test.js (hunk 3 above)
Comment (piperjarrett) [reply test/User-test.js]: okay, I wasn't sure.

**Review by Jedeo — APPROVED** [review]
Great job @piperjarrett thank you for the PR

File: src/User.js (hunk 1 above)
Comment (piperjarrett) [reply src/User.js]: Okay that worked too and I could take out the if statement.

File: src/User.js (hunk 2 above)
Comment (piperjarrett) [reply src/User.js]: done

File: test/User-test.js (hunk 3 above)
Comment (piperjarrett) [reply test/User-test.js]: done

**Review by forsethnico — APPROVED** [review]
Looks great! I think there might be another way to calculate the date/ week part, but it looks good to move on from here as is

File: src/User.js [hunk 4]
```diff
@@ -37,14 +37,10 @@ class User {
     const userSleep = sleepData.filter((entry) => entry.userID === this.id);
     const startingDate = userSleep.find((entry) => entry.date === startDate);
     const startIndex = userSleep.indexOf(startingDate);
-    let newArr = userSleep.slice(startIndex);
+    let newArr = userSleep.slice(startIndex, startIndex + 7);
```
Comment (Jedeo) [inline src/User.js:40]: I learned something new today

**Review by Jedeo — APPROVED** [review]

**Pull request merged**